src/main/resources/Confluence/Macros/ConfluenceGallery.xml

```diff
@@ -0,0 +1,723 @@
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </id>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Macro name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+## Get the parameters and put them in lists if necessary
+#set($title = $wikimacro.parameters.title)
+#set($includeString = $wikimacro.parameters.include)
+#set($excludeString = $wikimacro.parameters.exclude)
+#set($pagesString = $wikimacro.parameters.page)
+#set($include = $stringtool.split($includeString, ","))
+#set($exclude = $stringtool.split($excludeString, ","))
+#set($pages = $stringtool.split($pagesString, ","))
+## Display the gallery title if it is set (newlines are needed for the following macro calls to be standalone)
+#if(!$stringtool.isEmpty($title))
+    ==$title==
+    
+    
+#end
+## add the current page to the page list if no pages were given
+#if($stringtool.isEmpty($pagesString))
+    #set ($pages = [$doc.getTitle()])
+#end
+## Collect the attachments to display
+#set ($attachments = [])
+#foreach($page in $pages)
+    ## Resolve page and put attachments in temporary list
+    #set ($tempAttachments = [])
+    #set($query = $services.query.xwql("where doc.title = :name").bindValue('name', ''+${page}))
+    #set($pageStrings = $query.execute())
+    #if ($pageStrings.size() &gt; 0)
+        #set($pageDocument = $xwiki.getDocument($pageStrings[0]))
```

Comment:
This code doesn't seem to handle multiple pages in the `page` parameter while the description says `Pages containing the images to display.` Which of these is correct?

Reply:
It does handle multiple pages. The page parameter ($pagesString) is split on commas ($pages) and then the macro iterates over the entries in the $pages list.

Reply:
Indeed, the code is getting all the documents having a given title, hoping there is only one.